.github/workflows/release.yml

```diff
@@ -0,0 +1,58 @@
+name: aiodynamo release workflow
+
+on:
+  release:
+    types: [released] # only triggered on release, not pre-release or draft release
+
+permissions:
+  contents: read
+  id-token: write
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+defaults:
+  run:
+    shell: bash
+
+env:
+  PYTHON_VERSION_MIN: 3.8
+  TARGET_BRANCH: ${{ github.event.release.target_commitish }}
+  DEFAULT_BRANCH: ${{ github.event.repository.default_branch }}
+  TAG_NAME: ${{ github.ref_name }}
+
+jobs:
+  release:
+    name: Release to PyPI
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check whether release target branch is default branch
+        if: env.TARGET_BRANCH != env.DEFAULT_BRANCH
+        uses: actions/github-script@v6
+        with:
+          script: |
+            core.setFailed('Target branch (${{ env.TARGET_BRANCH }}) is not default branch (${{ env.DEFAULT_BRANCH }}).')
+      
+      - uses: actions/checkout@v3
+      
+      - uses: actions/setup-python@v4
+        with:
+          python-version: ${{ env.PYTHON_VERSION_MIN }}
```

Comment:
I'm curious about the rationale for using older Python.
Is it intended to ensure that the built artefacts actually work with older Python when they are downloaded from PYPI?
In that case, running tests would be better.

Otherwise, all this does is sets up what Python is used to run poetry.
So, why not 3.11?

Reply:
Since poetry is used for building, I just thought using the lowest possible python is better?
As set in the pyproject.toml
https://github.com/HENNGE/aiodynamo/blob/master/pyproject.toml#L24

Although I'm not that too familiar with how poetry performs the build process, so any input is appreciated.

Reply:
we might actually run into issues with this since we don't pin poetry and poetry drops support for older python versions once they get EOL'ed, so in about a year this will likely break. Using the highest version here might be better.